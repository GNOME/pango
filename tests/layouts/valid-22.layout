{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "comment" : "",
  "text" : "e0 = ooo...",
  "attributes" : [
    {
      "start" : 1,
      "end" : 2,
      "type" : "font-desc",
      "value" : "Italic"
    },
    {
      "start" : 1,
      "end" : 2,
      "type" : "font-scale",
      "value" : "subscript"
    },
    {
      "start" : 1,
      "end" : 2,
      "type" : "baseline-shift",
      "value" : "subscript"
    },
    {
      "start" : 6,
      "end" : 11,
      "type" : "font-scale",
      "value" : "superscript"
    },
    {
      "start" : 6,
      "end" : 11,
      "type" : "baseline-shift",
      "value" : "superscript"
    },
    {
      "start" : 7,
      "end" : 11,
      "type" : "font-scale",
      "value" : "superscript"
    },
    {
      "start" : 7,
      "end" : 11,
      "type" : "baseline-shift",
      "value" : "superscript"
    },
    {
      "start" : 8,
      "end" : 11,
      "type" : "font-scale",
      "value" : "superscript"
    },
    {
      "start" : 8,
      "end" : 11,
      "type" : "baseline-shift",
      "value" : "superscript"
    },
    {
      "start" : 9,
      "end" : 11,
      "type" : "font-scale",
      "value" : "superscript"
    },
    {
      "start" : 9,
      "end" : 11,
      "type" : "baseline-shift",
      "value" : "superscript"
    },
    {
      "start" : 10,
      "end" : 11,
      "type" : "font-scale",
      "value" : "superscript"
    },
    {
      "start" : 10,
      "end" : 11,
      "type" : "baseline-shift",
      "value" : "superscript"
    }
  ],
  "font" : "Cantarell 11",
  "output" : {
    "is-wrapped" : false,
    "is-ellipsized" : false,
    "unknown-glyphs" : 0,
    "width" : 49152,
    "height" : 22658,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-end" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 11,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 1,
            "text" : "e",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 1,
            "length" : 1,
            "text" : "0",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 7.1484375",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 1,
                "end" : 2,
                "type" : "font-scale",
                "value" : "subscript"
              },
              {
                "start" : 1,
                "end" : 2,
                "type" : "baseline-shift",
                "value" : "subscript"
              }
            ],
            "y-offset" : -3274,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 964,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 2,
            "length" : 4,
            "text" : " = o",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 1147,
                "width" : 9216,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 3
              }
            ]
          },
          {
            "offset" : 6,
            "length" : 1,
            "text" : "o",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 7.1484375",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 6,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 6,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              }
            ],
            "y-offset" : 5542,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 370,
                "width" : 5120,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 7,
            "length" : 1,
            "text" : "o",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 4.646484375",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 6,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 6,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              }
            ],
            "y-offset" : 9144,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 370,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 8,
            "length" : 1,
            "text" : ".",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 3.0205078125",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 6,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 6,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              }
            ],
            "y-offset" : 11485,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1058,
                "width" : 1024,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 9,
            "length" : 1,
            "text" : ".",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 1.962890625",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 6,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 6,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 9,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 9,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              }
            ],
            "y-offset" : 13007,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1058,
                "width" : 1024,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 10,
            "length" : 1,
            "text" : ".",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 1.2763671875",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 6,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 6,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 7,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 8,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 9,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 9,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              },
              {
                "start" : 10,
                "end" : 11,
                "type" : "font-scale",
                "value" : "superscript"
              },
              {
                "start" : 10,
                "end" : 11,
                "type" : "baseline-shift",
                "value" : "superscript"
              }
            ],
            "y-offset" : 13996,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1058,
                "width" : 0,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          }
        ]
      }
    ]
  }
}
