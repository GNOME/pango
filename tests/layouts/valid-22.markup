
e<span font_desc="italic"><sub>0</sub></span> = o<sup>o<sup>o<sup>.<sup>.<sup>.</sup></sup></sup></sup></sup>
