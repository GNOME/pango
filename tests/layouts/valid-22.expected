e0 = ooo...

--- parameters

wrapped: 0
ellipsized: 0
lines: 2

--- attributes

range 0 1
range 1 2
[1,2]font-desc=Italic
[1,2]font-scale=2
[1,2]baseline-shift=2
range 2 6
range 6 7
[6,11]font-scale=1
[6,11]baseline-shift=1
range 7 8
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
range 8 9
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
range 9 10
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
[9,11]font-scale=1
[9,11]baseline-shift=1
range 10 11
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
[9,11]font-scale=1
[9,11]baseline-shift=1
[10,11]font-scale=1
[10,11]baseline-shift=1
range 11 2147483647

--- directions

0 0 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 9(0) 10(1) 12(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'e0 = ooo...
'
i=2, index=12, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'e'
i=2, index=1, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, '0'
[1,2]font-scale=2
[1,2]baseline-shift=2
i=3, index=2, chars=4, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, ' = o'
i=4, index=6, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'o'
[6,11]font-scale=1
[6,11]baseline-shift=1
i=5, index=7, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'o'
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
i=6, index=8, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, '.'
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
i=7, index=9, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, '.'
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
[9,11]font-scale=1
[9,11]baseline-shift=1
i=8, index=10, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, '.'
[6,11]font-scale=1
[6,11]baseline-shift=1
[7,11]font-scale=1
[7,11]baseline-shift=1
[8,11]font-scale=1
[8,11]baseline-shift=1
[9,11]font-scale=1
[9,11]baseline-shift=1
[10,11]font-scale=1
[10,11]baseline-shift=1
i=9, index=11, no run, line end
i=10, index=12, no run, line end
